java-frontend/src/main/java/org/sonar/java/model/VisitorsBridge.java

```diff
@@ -79,7 +81,7 @@ public VisitorsBridge(Iterable visitors, List<File> projectClasspath, @Nullable
     this.scanners = scannersBuilder.build();
     this.executableScanners = scanners;
     this.sonarComponents = sonarComponents;
-    this.projectClasspath = projectClasspath;
+    this.classLoader = ClassLoaderBuilder.create(projectClasspath);
```

Comment:
Why not directly using a `SquidClassLoader` for the type of the `classLoader` field? (And consequently directly cast the result of the method invocation to it)

I know that it's using an implementation class rather than the interface, but our `ClassLoaderBuilder` only produces our own class loaders, and it would avoid the `instanceof SquidClassLoader` occurring at the end of the analysis. Its false branch is not covered nor coverable.